### PR TITLE
add examples support for draft info

### PR DESCRIPTION
**Pull request by davidgamero**
# Description

Add additional information for variable suggested example values in draft.yaml

These are added to enrich `draft info` output

- rename "info integration test" to "info e2e test" since it doesn't integrate with anything

Please note the following is a breaking change:
Update the `draft info` response to fit the following json format that allows example values and better captures the intent in JSON
```
{
  "supportedLanguages": [
    {
      "name": "clojure",
      "displayName": "Clojure",
      "variableExampleValues": {
        "VERSION": [
          "8-jdk-alpine",
          "11-jdk-alpine"
        ]
      }
    }
  ],
  "supportedDeploymentTypes": [
    "helm",
    "kustomize",
    "manifests"
  ]
}
```

Feature #164 

## Type of change
Feature

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [x] Updated e2e test for draft info command to validate new contract
- [x] Existing e2e and unit tests pass

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/config/draftconfig.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 
 //TODO: remove Name Overrides since we don't need them anymore
 type DraftConfig struct {
+	DisplayName      string              `yaml:"displayName"`
 	NameOverrides    []FileNameOverride  `yaml:"nameOverrides"`
```
Comment (jaiveerk) [inline pkg/config/draftconfig.go:10]: would removing NameOverrides be a larger change at this point than just deleting it from the struct definition?

File: template/dockerfiles/csharp/draft.yaml [hunk 2]
```diff
@@ -1,11 +1,13 @@
 language: csharp
+displayName: C#
 variables:
   - name: "PORT"
     description: "the port exposed in the application"
     type: int
   - name: "VERSION"
     description: "the dotnet SDK version"
     type: float
+    exampleValues: ["3.1","4.0","5.0","6.0"]
```
Comment (jaiveerk) [inline template/dockerfiles/csharp/draft.yaml:10]: super nit: some example values are ascending and some are descending, would we be able to standardize order?

(personal preference is for ascending but don't think it matters as long as it's consistent)

File: test/info_schema.json [hunk 3]
```diff
@@ -1,36 +1,107 @@
...
+          }
+        }
       }
     },
-    "supported_deployment_types": {
-      "$id": "#root/supported_deployment_types",
-      "title": "Supported_deployment_types",
+    "supportedDeploymentTypes": {
+      "$id": "#root/supportedDeploymentTypes",
+      "title": "Supporteddeploymenttypes",
```
Comment (jaiveerk) [inline test/info_schema.json:94]: nit: can we make this camelcase too?

**Review by jaiveerk — CHANGES_REQUESTED** [review]

File: test/check_info_schema.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+./draft info > ./info.json
+echo "Draft Info JSON schema:"
+cat ./test/info_schema.json
+  echo "Draft Info JSON:"
```
Comment (Tatsinnit) [inline test/check_info_schema.sh:6]: 🐟  nit: spaces for this `echo`

**Review by Tatsinnit — DISMISSED** [review]
❤️ 🙏 Thank you for adding me to this review, most of these are minor structural changes and `yaml` changes and some of them I have minor suggestions or question, hence added those in this PR. 

Rest Looks good to me. thank you.

File: template/dockerfiles/csharp/draft.yaml (hunk 2 above)
Comment (davidgamero) [reply template/dockerfiles/csharp/draft.yaml:10]: np i went with ascending

**Review by jaiveerk — APPROVED** [review]

**Review by qpetraroia — APPROVED** [review]

**Review by OliverMKing — APPROVED** [review]

File: test/info_schema.json [hunk 5]
```diff
@@ -1,36 +1,107 @@
...
       "type": "array",
       "default": [],
       "items":{
-        "$id": "#root/supported_deployment_types/items",
+        "$id": "#root/supportedDeploymentTypes/items",
```
[the remark this answers is not in the document]
Comment (Tatsinnit) [reply test/info_schema.json:98]: Thanks, Sounds good. FYI @qpetraroia I will let you decide if this is a breaking change and we need some information handling aka pinning issue at a repo level for the information regarding old schema will not work with new release et. al. hence reopening this for your eyes. Rest looks cool, thanks. ☕️❤️

**Review by Tatsinnit — APPROVED** [review]

**Pull request merged**